workbase/src/renderer/components/SchemaDesign/store/actions.js

```diff
@@ -211,10 +223,17 @@ export default {
 
         const node = state.visFacade.getNode(state.selectedNodes[0].id);
 
+        const edges = [];
         await Promise.all(payload.attributeTypes.map(async (attributeType) => {
-          const dataType = await (await graknTx.getSchemaConcept(attributeType)).dataType();
+          const type = await graknTx.getSchemaConcept(attributeType);
+          const dataType = await type.dataType();
           node.attributes = [...node.attributes, { type: attributeType, dataType }];
+
+          edges.push({ from: state.selectedNodes[0].id, to: type.id, label: 'has' });
```

Comment:
this is ok, but for future reference you can also do:

```
const edges = await Promise.all(
    .
    .
   return { from: state.selectedNodes[0].id, to: type.id,  label: 'has'}
)
```

I am adding this comment because it's a refactoring PR, so tips and trick for JS here and there 😄

Reply:
this is because `.map()` returns an array already